### PR TITLE
Add toggling of NetP Notifications to iOS

### PR DESCRIPTION
**Pull request by graeme**
Task/Issue URL: https://app.asana.com/0/0/1205768725084797/f
BSK: https://github.com/duckduckgo/BrowserServicesKit/pull/541

**Description**:

This adds the implementation of the VPN Notifications screen. It serves as a way to toggle NetP notifications on the iOS Network Extension and also to help prompt enabling notifications app-wide.

This is quite a design-heavy project, but there’s also use of an app-group shared UserDefaults store to share the toggle state between the status view, app side and the notifications presenter, extension side.

**Steps to test this PR**:
1. Delete the iOS app and reinstall (to reset notifications settings)
2. Launch and enable internal testing (Settings -> Debug -> Internal User)
3. Go to the Network Protection screen (Settings -> Network Protection)
4. Select VPN Notifications
5. Check it matches[ these designs](https://app.asana.com/0/0/1205739632361534/f)
6. Press Turn on Notifications here **and disallow them**
7. Press the button again which will take you to the iOS DuckDuckGo notifications settings. Enabled them.
8. Go back to DuckDuckGo to the VPN Notifications screen and check they match these (other) [designs ](https://www.figma.com/file/d7d8ZGkWD0W5ExhBI0tESv/O-N---VPN-Settings-Templates?type=design&node-id=564-19005&mode=design&t=xGigSHUt6PJkyaR9-0)
9. Go back to the NetP Status view. Enabled NetP. **Observe the notification**
10. Go to VPN Notifications one more time.
11. Go back to the NetP Status view. Disable NeP and reenable it. **Observe there is no notification.**
12. Extra: you can try the same with the test notifications from the Debug menu. They should also be affected by this toggle.

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `’`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/NetworkProtectionStatusView.swift [hunk 1]
```diff
@@ -40,9 +40,12 @@ struct NetworkProtectionStatusView: View {
             }
             settings()
         }
-        .animation(.default, value: statusModel.shouldShowError)
         .padding(.top, statusModel.error == nil ? 0 : -20)
-        .animation(.default, value: statusModel.shouldShowConnectionDetails)
+        .if(statusModel.animationsOn, transform: {
```
Comment (graeme) [inline DuckDuckGo/NetworkProtectionStatusView.swift:44]: The animation bug this fixes became a lot more noticeable with the addition of the new section. So I fixed it here.

File: Core/UserDefaults+NetworkProtection.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import Foundation
+
+public extension UserDefaults {
+    static var networkProtectionGroupDefaults: UserDefaults {
+        let suiteName = "\(Global.groupIdPrefix).netp"
+        guard let defaults = UserDefaults(suiteName: suiteName) else {
+            fatalError("Failed to create netP UserDefaults")
+        }
+        return defaults
+    }
```
Comment (samsymons) [inline Core/UserDefaults+NetworkProtection.swift:31]: Nice, I suspect I'll be using this for the widget - thanks!

File: DuckDuckGo/UIApplicationExtension.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        let settingsString: String
+        if #available(iOS 16, *) {
+            settingsString = UIApplication.openNotificationSettingsURLString
+        } else if #available(iOS 15.4, *) {
+            settingsString = UIApplicationOpenNotificationSettingsURLString
+        } else {
+            settingsString = UIApplication.openSettingsURLString
+        }
+        return URL(string: settingsString)
+    }()
```
Comment (samsymons) [inline DuckDuckGo/UIApplicationExtension.swift:33]: 👍

File: DuckDuckGo/NotificationsAuthorizationController.swift
Comment (samsymons) [inline DuckDuckGo/NotificationsAuthorizationController.swift:1]: Nice. We have some notification handling code that exists for the waitlist, but I don't think it's as complete as this - I'll keep this in mind for the next waitlist.

File: DuckDuckGo/NetworkProtectionVPNNotificationsView.swift [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @ViewBuilder
+    private var authorizedView: some View {
+        Section {
+            Toggle(UserText.netPVPNAlertsToggleTitle, isOn: Binding(
+                get: { model.alertsEnabled },
+                set: model.didToggleAlerts(to:)
+            ))
+            .toggleStyle(SwitchToggleStyle(tint: Color(designSystemColor: .accent)))
```
Comment (samsymons) [inline DuckDuckGo/NetworkProtectionVPNNotificationsView.swift]:
```suggestion
            .toggleStyle(SwitchToggleStyle(tint: .controlColor))
```

**Review by samsymons — APPROVED** [review]
Looks great, and works as expected - nice one!

**Pull request merged**